packages/calcite-components/src/components/list/list.tsx

```diff
@@ -397,15 +394,45 @@ export class List implements InteractiveComponent, LoadableComponent {
     }
   };
 
-  handleFilter = (event: CustomEvent): void => {
+  private updateFilteredData(emit = false): void {
+    const { filterEl } = this;
+
+    if (!filterEl) {
+      return;
+    }
+
+    if (filterEl.filteredItems) {
+      this.filteredData = filterEl.filteredItems as ItemData;
+    }
+
+    this.updateListItems(emit);
+  }
+
+  private async performFilter(): Promise<void> {
```

Comment:
Some naming suggestions: `filter()`,  `runFilter`, `refreshFilter()`, `updateFilter`? Up to you.